tests/platform_helper/utils/test_validation.py

```diff
@@ -940,58 +940,66 @@ def test_validate_database_copy_multi_postgres_failures(capfd):
     )
 
 
+@pytest.mark.parametrize(
+    "config, expected_response",
+    [
+        (
+            # No engine defined in either env
+            {
+                "extensions": {
+                    "connors-redis": {
+                        "type": "redis",
+                        "environments": {"*": {"plan": "tiny"}, "prod": {"plan": "largish"}},
+                    }
+                },
+            },
+            "",
+        ),
+        (
+            # Valid engine version defined in *
+            {
+                "extensions": {
+                    "connors-redis": {
+                        "type": "redis",
+                        "environments": {
+                            "*": {"engine": "7.1", "plan": "tiny"},
+                            "prod": {"plan": "tiny"},
+                        },
+                    }
+                },
+            },
+            "",
+        ),
+        (
+            # Invalid engine defined in prod environment
+            {
+                "extensions": {
+                    "connors-redis": {
+                        "type": "redis",
+                        "environments": {
+                            "*": {"plan": "tiny"},
+                            "prod": {"engine": "6.1", "plan": "tiny"},
+                        },
+                    }
+                },
+            },
+            "redis version for environment prod is not in the list of supported redis versions: ['7.1']. Provided Version: 6.1",
+        ),
+    ],
+)
 @patch("dbt_platform_helper.utils.validation.get_supported_redis_versions", return_value=["7.1"])
-def test_validate_extensions_supported_versions_successful_with_supported_version(
-    mock_supported_versions, capsys
+def test_validate_extensions_different_scenarios(
```

Comment:
The different scenarios bit is covered by the parameterisation.
```suggestion
def test_validate_extension_supported_versions(
```